pkg/streamingpromql/query.go

```diff
@@ -310,10 +313,11 @@ func (q *Query) convertToInstantVectorOperator(expr parser.Expr, timeRange types
 }
 
 func (q *Query) convertFunctionCallToInstantVectorOperator(e *parser.Call, timeRange types.QueryTimeRange) (types.InstantVectorOperator, error) {
-	// Handle special toggles for classic histograms
-	if !q.engine.featureToggles.EnableHistogramQuantileFunction {
-		if e.Func.Name == "histogram_quantile" {
-			return nil, compat.NewNotSupportedError(fmt.Sprintf("'%s' function", e.Func.Name))
+	for _, fName := range q.engine.mqeOpts.DisabledFunctions {
+		// e.Func.Name is already validated by the parser. Meaning we don't need to check if the function name
+		// is real before checking if it is disabled.
+		if e.Func.Name == fName {
+			return nil, compat.NewNotSupportedError(fmt.Sprintf("'%s' function is disabled", e.Func.Name))
```

Comment:
Using `slices.Contains` would make the intention clearer here, and if we sort `DisabledFunctions`, then we could use `slices.BinarySearch` to make this search faster as well.

Reply:
I'm not concerned about the overhead here as it's just a few operations per query in comparison to the data we're processing. That said, it's easy enough so will make the optimisations.